src/lib/php/Dao/LicenseDao.php

```diff
@@ -313,6 +314,147 @@ public function getLicenseIdPerPfileForAgentId(ItemTreeBounds $itemTreeBounds, $
 
   /**
    * @param ItemTreeBounds $itemTreeBounds
+   * @param Array(int) $selectedAgentIds
+   * @param bool $includeSubFolders
+   * @param Array(String) $nameRange
+   * @param string $excluding
+   * @param bool $ignore ignore files without license
+   * @return array
+   */
+  public function getLicensesPerFileNameForAgentId(ItemTreeBounds $itemTreeBounds,
+                                                   $selectedAgentIds=null,
+                                                   $includeSubfolders=true,
+                                                   $nameRange=array(),
+                                                   $excluding='',
+                                                   $ignore=0)
+  {
+    $uploadTreeTableName = $itemTreeBounds->getUploadTreeTableName();
+    $statementName = __METHOD__ . '.' . $uploadTreeTableName;
+    $param = array();
+
+    if ($includeSubfolders)
+    {
+      $param[] = $itemTreeBounds->getLeft();
+      $param[] = $itemTreeBounds->getRight();
+      $condition = " lft BETWEEN $1 AND $2";
+      $condition .= " AND (ufile_mode & (1<<28)) = 0";
```

Comment:
This condition means 'ufile is no artifact'. With same condition in else case, the check in [line 413](https://github.com/fossology/fossology/pull/634/files#diff-691d17f7359c7e2f3074a185349fa863R413) could be avoided.